gui/webdisplay/src/TWebWindow.cxx

```diff
@@ -326,6 +343,18 @@ std::string ROOT::Experimental::TWebWindow::RelativeAddr(std::shared_ptr<TWebWin
    return res;
 }
 
+///////////////////////////////////////////////////////////////////////////////////
+/// returns connection for specified connection number
+/// Total number of connections can be retrieved with NumConnections() method
+
+unsigned ROOT::Experimental::TWebWindow::GetConnectionId(unsigned num) const
+{
+   for (auto iter = fConn.begin(); iter != fConn.end(); ++iter) {
+      if (num-- == 0) return iter->fConnId;
```

Comment:
Isn't this 
```
auto iter = fConn.begin() + num;
return iter->fConnId;
```
?

Reply:
This code includes safety check - if num too large, it returns 0.
Simple case - when container empty

Reply:
But then my version gives you the chance to make that check explicit, and have an explicit error message for this case.